internal/terraform/context_plan.go

```diff
@@ -574,7 +574,7 @@ func (c *Context) planGraph(config *configs.Config, prevRunState *states.State,
 		}).Build(addrs.RootModuleInstance)
 		return graph, walkPlan, diags
 	case plans.DestroyMode:
-		graph, diags := (&DestroyPlanGraphBuilder{
+		graph, diags := DestroyPlanGraphBuilder(&PlanGraphBuilder{
```

Comment:
Would it be reasonable to make this be consistent with how we deal with `RefreshOnlyMode` above and have this instead "just" set a flag in the `PlanGraphBuilder` struct and let the `Build` method worry about the details?

I think I ended up leaving it in this odd shape where `NormalMode` and `RefreshOnlyMode` are _largely_ the same just because the `DestroyMode` case was a drastically different shape, but if we normalize these to all just be conditional flags on the struct then we could potentially rework this to avoid duplicating the common fields for each case and _only_ set `skipPlanChanges` / `destroy` in the non-normal modes.

Of course fine to leave this for another day if it seems too risky, since it's just some further cleanup after all.